lib/services/oauth/model.go

```diff
@@ -0,0 +1,38 @@
+package oauth
+
+import "time"
+
+// Client represents an OAuth client's model in the db
+type Client struct {
+	ID          int    `db:"id"`
+	ExternalID  string `db:"external_id"`
+	Secret      string `db:"secret"`
+	RedirectURI string `db:"redirect_uri"`
+	UserData    string `db:"user_data"`
+}
+
+// Authorization represents an OAuth client's authorization (who they are)
+type Authorization struct {
+	ID          int       `db:"id"`
+	ClientID    int       `db:"client_id"`
+	Code        string    `db:"code"`
+	Expiration  int32     `db:"expiration"`
+	Scope       string    `db:"scope"`
+	RedirectURI string    `db:"redirect_uri"`
+	StateData   string    `db:"state_data"`
+	CreatedAt   time.Time `db:"created_at"`
+	UserData    string    `db:"user_data"`
```

Comment:
Is this where we will store data about what user ID this authorization corresponds to?

Reply:
Yeah, `osin` has a https://godoc.org/github.com/RangelReale/osin#Storage interface that we can implement, that the CRUD here should help us with.